packages/router/src/page-loader.tsx

```diff
@@ -4,7 +4,12 @@ import isEqual from 'lodash.isequal'
 
 import { createNamedContext, ParamsContext, Spec } from './internal'
 
-export const PageLoadingContext = createNamedContext('PageLoading')
+export const PageLoadingContext = createNamedContext<{ loading: boolean }>(
```

Comment:
Can you please create a type for the context and export that type?
It's how we usually do typed contexts.

Reply:
That seems odd, not using inference for when it's available? I could extract it, but for one value it seems odd

Reply:
The reson to make an interface for it is so it can be referenced by name. That's also why it should be exported -- so others can use it. Maybe that's not needed here, but it's a pattern I like to stick to with all contexts I create.

Reply:
Done